### PR TITLE
add support for weapp

### PR DESCRIPTION
**Pull request by taoqf**
For now, if using with webpack, need wait for https://github.com/jryans/timers-browserify/pull/24 and https://github.com/maxogden/websocket-stream/pull/129 merged.

### CODE REVIEW

**Comment by mcollina** [comment]
I'm a bit lost in what is weapp, wx and such.

**Comment by taoqf** [comment]
https://mp.weixin.qq.com/cgi-bin/wx
I have added something to the readme,but tests could not be added because it's only could run in container.

**Comment by taoqf** [comment]
@mcollina What can I do with the failled checks?

**Comment by taoqf** [comment]
@mcollina Weixin(wx) is an application runs on android, iPhone, you may think it as [cordova](http://cordova.apache.org/), It allow h5 application runs in it, and it allow user use WebSocket using it's [api](https://mp.weixin.qq.com/debug/wxadoc/dev/api/network-socket.html).

**Comment by taoqf** [comment]
Then could I add some unit test with failing judge? I think there are so many chinese programmer would need this.

File: .jshintrc [hunk 1]
```diff
@@ -17,6 +17,5 @@
   "mocha": true,
   "indent": 2,
   "latedef": true,
-  "immed": true,
```
Comment (mcollina) [inline .jshintrc]: why this?

Comment (taoqf) [inline .jshintrc]: it was defined multi times. see line 4

**Comment by mcollina** [comment]
Unit tests will be good?

**Comment by taoqf** [comment]
I don't know how to add unit test yet, but I would try my best.

**Comment by taoqf** [comment]
I have done some test, not unit test, it works good.

**Comment by taoqf** [comment]
@mcollina so need I add unit test for this pr?

**Comment by mcollina** [comment]
Yes.

**Comment by taoqf** [comment]
Hi, @mcollina , I have add some test into `/test/wx.js` and passed `npm run weapp-test`.
is this the right way to add unit test?

**Comment by taoqf** [comment]
Whatever I try, still could pass `codecov` check. any help?

**Comment by mcollina** [comment]
I'll merge this. However, can you please restore `lib/connect/index.js` to what is there on master? It's all reported as changed in this PR.

File: lib/connect/index.js [hunk 2]
```diff
@@ -1,146 +1,153 @@
...
+if (process.title !== 'browser') {
+  protocols.mqtt = require('./tcp')
+  protocols.tcp = require('./tcp')
+  protocols.ssl = require('./tls')
+  protocols.tls = require('./tls')
+  protocols.mqtts = require('./tls')
+}
+
+protocols.wx = require('./wx')
+protocols.wxs = require('./wx')
```
Comment (mcollina) [inline lib/connect/index.js]: we should not be adding `wx` if it's not a browser. the main `wx` object is not there.

**Comment by taoqf** [comment]
@mcollina , Thanks, I move the code into `else` block, and I skiped the unit tests I added because it could not be running on node.js.

**Comment by mcollina** [comment]
You can remove the skipped tests.

What have you changed to the lib/connect/index.js file? GH reports it was all changed, but it does not seems so. Can you please restore it to what was before and apply only your modifications?

**Comment by taoqf** [comment]
I tried not skip the tests and it will cause a failing test after that test.
```bash
121 passing (16s)
  1 failing

  1) mqtt #connect should return an MqttClient when connect is called with wx:/ url:
     AssertionError: expected Function { name: '' } to throw exception
      at Assertion.fail (node_modules/should/cjs/should.js:258:17)
      at Assertion.value (node_modules/should/cjs/should.js:335:19)
      at Context.<anonymous> (test/mqtt.js:116:22)
```
I don't really know what was wrong with github, I had not changed the whole file `connect/index.js`. 
I will try to create a new pr later and see if gh reports the same.

**Comment by vikingsailor** [comment]
@taoqf 在微信小程序里怎么使用mqtt连接AZURE?我已经在nodejs中联通了iot云，但是想在小程序中看到各设备的消息？如何实现？

**Comment by taoqf** [comment]
Could you please describe azure?

**Pull request closed without merging**